### PR TITLE
LEP-279 tighten validation in v6 controller

### PR DESCRIPTION
**Pull request by starswan**
https://dsdmoj.atlassian.net/browse/LEP-279

Add tightened validation from v7 controller into v6

### CODE REVIEW

File: spec/requests/swagger_docs/v6/full_assessment_spec.rb [hunk 1]
```diff
@@ -108,26 +112,7 @@
...
-                              },
-                              client_id: {
-                                type: :string,
-                                description: "Client supplied id to identify the employment",
-                              },
-                              payments: { "$ref" => components[:employment_payment_list] },
-                            },
-                          },
-                        },
+                        employments: { "$ref" => components[:employments] },
```
Comment (ripan) [inline spec/requests/swagger_docs/v6/full_assessment_spec.rb:115]: @starswan With this change we will be adding additional attribute `receiving_only_statutory_sick_or_maternity_pay` and i think this change is correct  😄

File: app/lib/swagger_docs.rb [hunk 2]
```diff
@@ -258,27 +258,43 @@ def schema
                   example: "1992-07-22",
                 },
                 gross: {
-                  "$ref" => SCHEMA_COMPONENTS[:currency],
-                  description: "Gross payment income received",
-                  example: "101.01",
+                  allOf: [
```
[the remark this answers is not in the document]
Comment (starswan) [reply app/lib/swagger_docs.rb:261]: The original was creating invalid schema documents in swagger.yaml ...

File: swagger/v6/swagger.yaml [hunk 3]
```diff
@@ -1700,25 +1734,7 @@ paths:
...
-                          name:
-                            type: string
-                            description: Identifying name for this employment - e.g.
-                              employer's name
-                          client_id:
-                            type: string
-                            description: Client supplied id to identify the employment
-                          payments:
-                            "$ref": "#/components/schemas/EmploymentPaymentList"
+                      "$ref": "#/components/schemas/Employments"
```
[the remark this answers is not in the document]
Comment (starswan) [reply swagger/v6/swagger.yaml:1737]: And that is correct? (its the generated version of the change above)

**Comment by davidread** [comment]
Slight risk when merging this - do first thing in the morning, so we are on hand to revert if needed

**Review by ripan — APPROVED** [review]

**Pull request merged**